test/test_shared_library.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <string>
+
+#include "./allocator_testing_utils.h"
+
+#include "rcutils/allocator.h"
+#include "rcutils/error_handling.h"
+#include "rcutils/shared_library.h"
+
+#include "rcutils/get_env.h"
+
+class TestSharedLibrary : public ::testing::Test
+{
+protected:
+  void SetUp() final
+  {
+    // Reset rcutil error global state in case a previously
+    // running test has failed.
+    rcutils_reset_error();
+    lib = rcutils_get_zero_initialized_shared_library();
+
+#ifdef __linux__
+    library_path = std::string("libdummy_shared_library.so");
+#elif __APPLE__
+    library_path = std::string("libdummy_shared_library.dylib");
+#elif _WIN32
+    library_path = std::string("dummy_shared_library.dll");
```

Comment:
For calling code having to know about these platform specific naming details makes the API rather difficult to use. A function to provide the platform specific library name when passing in this case `dummy_shared_library` would be helpful.

Reply:
Added a function to provide the platform specific library name https://github.com/ros2/rcutils/pull/215/commits/d941e44755bffda67808ca955b753c445cfcffb4